src/NServiceBus.AcceptanceTesting/ScenarioWithContext.cs

```diff
@@ -54,6 +55,17 @@ public Task<TContext> Run(TimeSpan? testExecutionTimeout = null)
             return WithEndpoint<T>(b => { });
         }
 
+        public IScenarioWithEndpointBehavior<TContext> WithCustomComponent(
+            string name,
+            Func<TContext, CancellationToken, Task> onStart, 
+            Func<TContext, Task> onStop, 
+            Func<TContext, CancellationToken, Task> onStarted = null) 
```

Comment:
it doesn't make much sense to enforce onStart/onStop to be provided while onStarted is optional. I think it would be more consistent if all params would be optional.